rubix-hadoop2/src/main/java/com/qubole/rubix/hadoop2/Hadoop2ClusterManager.java

```diff
@@ -0,0 +1,219 @@
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.hadoop2;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+
+import com.qubole.rubix.core.ClusterManager;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Created by sakshia on 28/7/16.
+ */
+public class Hadoop2ClusterManager
+        extends ClusterManager
+{
+
+    private boolean isMaster = true;
+    public int serverPort = 8088;
+    private String serverAddress = "localhost";
+    private Supplier<List<String>> nodesSupplier;
+    String address = "localhost:8088";
+    private static final Logger log = Logger.getLogger(Hadoop2ClusterManager.class.getName());
+
+    static String ADDRESS = "yarn.resourcemanager.webapp.address";
+
+    @Override
+    public void initialize(Configuration conf)
+    {
+        super.initialize(conf);
+        log.info("Initializing: Hadoop2ClusterManager");
+        this.address = conf.get(ADDRESS, address);
+        this.serverAddress = address.substring(0, address.indexOf(":"));
+        this.serverPort = Integer.parseInt(address.substring(address.indexOf(":") + 1));
+        nodesSupplier = Suppliers.memoizeWithExpiration(new Supplier<List<String>>()
+        {
+
+            @Override
+            public List<String> get()
+            {
+                if (!isMaster) {
+                    // First time all nodes start assuming themselves as master and down the line figure out their role
+                    // Next time onwards, only master will be fetching the list of nodes
+                    return ImmutableList.of();
+                }
+                try {
+                    StringBuffer response = new StringBuffer();
+                    URL obj = getNodeURL();
+                    HttpURLConnection httpcon = (HttpURLConnection) obj.openConnection();
```

Comment:
check the javadocs for HttpURLConnection, we should close the connections before returning